opm/common/data/SimulationDataContainer.cpp

```diff
@@ -75,6 +75,36 @@ namespace Opm {
     }
 
 
+    void SimulationDataContainer::setCellDataComponent( const std::string& key , size_t component , const std::vector<int>& cells , const std::vector<double>& values) {
+        auto& data = getCellData( key );
+        if (component >= m_num_phases)
```

Comment:
Why would we have the restriction that component < m_num_phases? I think the num_phases thing should be removed from this class. A more specific SimulatorState may add it, but I don't think it makes sense for this class.